commit 0a282de11b87fb4ad015c0cd10158555999705f9
Author: Arjun Shankar <arjun@redhat.com>
Date:   Wed Apr 7 00:30:11 2021 +0200

    malloc: Run tst-malloc-stats-cancellation via test-driver.c
    
    This allows the test to time out in case it hangs.
    
    Reviewed-by: Carlos O'Donell <carlos@redhat.com>

diff --git a/malloc/tst-malloc-stats-cancellation.c b/malloc/tst-malloc-stats-cancellation.c
index 9a8f475830b16af2..725cad6498f3cdf8 100644
--- a/malloc/tst-malloc-stats-cancellation.c
+++ b/malloc/tst-malloc-stats-cancellation.c
@@ -91,8 +91,8 @@ buffer_threadproc (void *argp)
 }
 
 
-int
-main (void)
+static int
+do_test (void)
 {
   int result = 0, err, real_stderr_fd, bufpipe[2];
   pthread_t t_thr, b_thr;
@@ -214,3 +214,5 @@ main (void)
     }
   return result;
 }
+
+#include <support/test-driver.c>
